apm-agent-api/pom.xml

```diff
@@ -15,7 +22,34 @@
     <name>${project.groupId}:${project.artifactId}</name>
 
     <build>
+        <resources>
+            <resource>
+                <directory>${apm-agent-parent.base.dir}</directory>
+                <includes>
+                    <include>LICENSE</include>
+                </includes>
+                <targetPath>META-INF</targetPath>
+            </resource>
+            <resource>
+                <directory>src/main/resources/META-INF</directory>
+                <targetPath>META-INF</targetPath>
+            </resource>
+        </resources>
         <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-source-plugin</artifactId>
+                <version>3.0.1</version>
+                <executions>
+                    <execution>
+                        <id>generate-source-jar</id>
```

Comment:
does that fix the shaded source jar?

Reply:
No, this one was created properly. The addition of this plugin in the parent project maybe took care of that, not sure. However, once I removed it from there, I had to add it here and to the opentracing module